frontend/rolecall/src/app/cast/cast-drag-and-drop.component.ts

```diff
@@ -242,8 +242,13 @@ export class CastDragAndDrop implements OnInit {
     this.data = [[]];
     this.cast = this.castAPI.castFromUUID(this.selectedCastUUID);
     this.ensureAllPositionsMet(this.cast);
-    this.positionVals = this.positionVals.sort((a, b) => { return a.name < b.name ? -1 : 1 });
-    this.cast.filled_positions = this.cast.filled_positions.filter(val => this.positionVals.find(val2 => val2.uuid == val.position_uuid)).sort((a, b) => { return this.positionVals.find(val => val.uuid == a.position_uuid).name < this.positionVals.find(val => val.uuid == b.position_uuid).name ? -1 : 1 });
+    // this.positionVals = this.positionVals.sort((a, b) => { return a.name < b.name ? -1 : 1 });
+    // YHE
+    this.positionVals = this.positionVals.sort((a, b) => { return a.order < b.order ? -1 : 1 });
```

Comment:
This is a little hard to read with the wrapping, consider splitting things up a bit like this:
```
this.cast.filled_positions = this.cast.filled_positions
    .filter(filledPosition => this.positionVals.find(val => val.uuid == filledPosition.position_uuid))
    .sort((a, b) => {
      let orderA = this.positionVals.find(val => val.uuid == a.position_uuid).order;
      let orderB = this.positionVals.find(val => val.uuid == b.position_uuid).order;
      return orderA < orderB ? -1 : 1;
    });
```

Reply:
Done.